lib/Scheduler/bb_spill.cpp

```diff
@@ -313,6 +303,42 @@ InstCount BBWithSpill::CmputCostLwrBound() {
 
   return staticLowerBound;
 }
+
+InstCount BBWithSpill::CmputExecCostLwrBound() {
+  ExecCostLwrBound_ = schedLwrBound_ * schedCostFactor_;
+  return ExecCostLwrBound_;
+}
+
+InstCount BBWithSpill::CmputRPCostLwrBound() {
+  InstCount spillCostLwrBound = 0;
+
+  if (GetSpillCostFunc() == SCF_SLIL) {
+    spillCostLwrBound =
+        ComputeSLILStaticLowerBound(regTypeCnt_, regFiles_, dataDepGraph_);
+    dynamicSlilLowerBound_ = spillCostLwrBound;
+    staticSlilLowerBound_ = spillCostLwrBound;
+  }
+
+  RpCostLwrBound_ = spillCostLwrBound * SCW_;
+  return RpCostLwrBound_;
+}
+/*****************************************************************************/
+
+void BBWithSpill::addRecordedCost(SPILL_COST_FUNCTION Scf) {
+  // Do nothing if the CF is already recorded
+  for (auto it = recordedCostFunctions.begin();
+       it != recordedCostFunctions.end(); ++it)
+    if (*it == Scf)
+      return;
```

Comment:
[`llvm::is_contained()`](https://llvm.org/doxygen/namespacellvm.html#acd1cd968cb420c82d70926920fcdc7d7).

```
if (llvm::is_contained(recordedCostFunctions, Scf)) return;
```

Reply:
Thanks, I will swap out this code